### DISCUSSION TITLE
Is there a way to obtain a list of all the files I've downloaded?

### DISCUSSION THREAD

**Discussion by thenormal** [nicotine-plus/nicotine-plus#2475] [Q&A]
Hello everyone.

I have been using Nicotine+ for a few months now, and I love it. I have a question about it.

Some days ago I downloaded some files from a user. I accidentally delete them, and I have been trying to find a way to locate the user's username, so I can send them a message or get notified when they come back online.

I noticed there is a log that includes the downloads of the current day (when I am done downloading something, I just empty out the queue), but such log file basically resets itself every time the program is launched.

I wonder if there's a way to obtain a list of my previous downloads and related users, even if the downloaded files have been removed from the queue in the Download tab.

**Comment by slook** [comment]
You can enable `Preferences > Logging > "Log transfers to file"` which would do what you need. It constantly records data about all started transfers (Downloads and Uploads) including the file path, username, date and time.

The `transfers.log` file is persistent regardless of clearing the lists in the Downloads and Uploads tabs.

Unfortunately, since the option is not enabled by default to capture the log of historic transfers, it is not possible to recover data about your accidentally deleted downloads from before.

The only possible way might be to use data recovery software to restore previous versions of the `downloads.json` file to see if the username data you lost might still be buried in the filesystem.
